crates/volta-core/src/tool/mod.rs

```diff
@@ -193,6 +194,24 @@ impl ToolSpec {
     }
 }
 
+/// Function for comparing toolspecs for use in sorting.
+/// We want to preserve the original order as much as possible, so we treat same tool categories as equal
+/// We still need to pull Node up first, followed by Npm / Yarn, and then Packages last
+fn tool_spec_sort_comparator(left: &ToolSpec, right: &ToolSpec) -> Ordering {
```

Comment:
I'd suggest moving this to be an implementation on the `ToolSpec`:

```rust
impl ToolSpec {
    // ... all the other impls
    fn sort_comparator(left: &ToolSpec, right: &ToolSpec) -> Ordering {
        // ... the same body you have below
    }
}
```

 The invocation would then just be `tools.sort_by(Self::sort_comparator)`.

Reply:
Good thinking, that's a lot cleaner and better encapsulated!